skrub/_datetime_encoder.py

```diff
@@ -284,16 +333,50 @@ def fit_transform(self, column, y=None):
                 f"Column {sbd.name(column)!r} does not have Date or Datetime dtype."
             )
         if self.resolution is None:
-            self.extracted_features_ = []
+            self._partial_features = []
         else:
             idx_level = _TIME_LEVELS.index(self.resolution)
             if _is_date(column):
                 idx_level = min(idx_level, _TIME_LEVELS.index("day"))
-            self.extracted_features_ = _TIME_LEVELS[: idx_level + 1]
+            self._partial_features = _TIME_LEVELS[: idx_level + 1]
         if self.add_total_seconds:
-            self.extracted_features_.append("total_seconds")
+            self._partial_features.append("total_seconds")
         if self.add_weekday:
-            self.extracted_features_.append("weekday")
+            self._partial_features.append("weekday")
+        if self.add_day_of_year:
+            self._partial_features.append("day_of_year")
+
+        # Adding transformers for periodic encoding
+        self._required_transformers = {}
+
+        col_name = sbd.name(column)
+        self.extracted_features_ = [
+            f"{col_name}_{_feat}" for _feat in self._partial_features
+        ]
+
+        # Iterating over all attributes that end with _encoding to use the default
+        # parameters
+        if self.periodic_encoding is not None:
+            enc_attr = ["year", "month", "weekday", "hour"][: idx_level + 1]
```

Comment:
here aren't we encoding 

- the year when really want to be encoding the day_of_year (or the month with a period of 12)
- the month when we really want to be encoding the day of the month

or am I missing something?